src/main/webapp/gameHelper.js

```diff
@@ -179,3 +185,31 @@ function changeThreshold() {
   fetch('/promotion-threshold', {method: 'POST', body: params});
   window.location='promotionquiz.html';
 }
+
+/*
+ * This function deconstructs the current dialogue displayed and searches for PLAYER_NAME_PARAMETER
+ * Then replaces it with playerNickname
+ */
+function deconstructsSentence() {
```

Comment:
There is a string method called replace/ replaceAll that might do just what you're doing here in two methods in one line.